src/model/ThreeJSForwardTranslator.hpp

```diff
@@ -64,10 +64,16 @@ namespace model {
     /// Get error messages generated by the last translation.
     std::vector<LogMessage> errors() const;
 
+    // Check if Surface is Convex, and check if its correctly oriented given the containing space, whether the space itself is Convex, Enclosed
+    // This is computationally intensive, so it's opt-in
+    bool includeGeometryDiagnostics() const;
+    void setIncludeGeometryDiagnostics(bool includeGeometryDiagnostics);
```

Comment:
Optional include of geometry diagnostics though with caching it's not **that** bad.

before caching:

| Test case                  | Opt-out | Opt-out | Opt-in | Opt-in  |
|----------------------------|---------|---------|--------|---------|
|                            | Debug   | Release | Debug  | Release |
| 7_7_Windows_Complete       | 84.63   | 11.78   | 26.76  | 3.68    |
| ParkUnder_Retail_Office_C2 | 120.64  | 18.08   | 15.92  | 2.13    |


**After caching, in release:**

```
1/4 Test #2562: ModelFixture.ThreeJSForwardTranslator_RefBldgOutPatientNew2004_Chicago .......................   Passed    1.39 sec
2/4 Test #2563: ModelFixture.ThreeJSForwardTranslator_RefBldgOutPatientNew2004_Chicago_GeometryDiagnostics ...   Passed    1.40 sec
3/4 Test #2560: ModelFixture.ThreeJSForwardTranslator_7_7_Windows_Complete ...................................   Passed    3.79 sec
4/4 Test #2561: ModelFixture.ThreeJSForwardTranslator_7_7_Windows_Complete_GeometryDiagnostics ...............   Passed    3.95 sec
```

**After caching, in debug:**

```
1/4 Test #2562: ModelFixture.ThreeJSForwardTranslator_RefBldgOutPatientNew2004_Chicago .......................   Passed   10.09 sec
2/4 Test #2563: ModelFixture.ThreeJSForwardTranslator_RefBldgOutPatientNew2004_Chicago_GeometryDiagnostics ...   Passed   10.19 sec
3/4 Test #2560: ModelFixture.ThreeJSForwardTranslator_7_7_Windows_Complete ...................................   Passed   27.17 sec
4/4 Test #2561: ModelFixture.ThreeJSForwardTranslator_7_7_Windows_Complete_GeometryDiagnostics ...............   Passed   28.33 sec
```